packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -132,101 +129,49 @@ function InsertionPointPopover( {
 }
 
 export default function InsertionPoint( { children, containerRef } ) {
-	const [ isInserterShown, setIsInserterShown ] = useState( false );
 	const [ isInserterForced, setIsInserterForced ] = useState( false );
-	const [ inserterClientId, setInserterClientId ] = useState( null );
-	const { isMultiSelecting, isInserterVisible, selectedClientId } = useSelect(
-		( select ) => {
-			const {
-				isMultiSelecting: _isMultiSelecting,
-				isBlockInsertionPointVisible,
-				getBlockInsertionPoint,
-				getBlockOrder,
-			} = select( 'core/block-editor' );
-
-			const insertionPoint = getBlockInsertionPoint();
-			const order = getBlockOrder( insertionPoint.rootClientId );
-
-			return {
-				isMultiSelecting: _isMultiSelecting(),
-				isInserterVisible: isBlockInsertionPointVisible(),
-				selectedClientId: order[ insertionPoint.index ],
-			};
-		},
-		[]
-	);
-
-	function onMouseMove( event ) {
-		if (
-			! event.target.classList.contains(
-				'block-editor-block-list__layout'
-			)
-		) {
-			if ( isInserterShown ) {
-				setIsInserterShown( false );
-			}
-			return;
-		}
-
-		const rect = event.target.getBoundingClientRect();
-		const offset = event.clientY - rect.top;
-		const element = Array.from( event.target.children ).find(
-			( blockEl ) => {
-				return blockEl.offsetTop > offset;
-			}
-		);
-
-		if ( ! element ) {
-			return;
-		}
-
-		const clientId = element.id.slice( 'block-'.length );
-
-		if ( ! clientId ) {
-			return;
-		}
-
-		const elementRect = element.getBoundingClientRect();
-
-		if (
-			event.clientX > elementRect.right ||
-			event.clientX < elementRect.left
-		) {
-			if ( isInserterShown ) {
-				setIsInserterShown( false );
-			}
-			return;
-		}
-
-		setIsInserterShown( true );
-		setInserterClientId( clientId );
-	}
-
-	const isVisible = isInserterShown || isInserterForced || isInserterVisible;
+	const {
+		isMultiSelecting,
+		isInserterVisible,
+		selectedClientId,
+		isNestedBlock,
+	} = useSelect( ( select ) => {
+		const {
+			isMultiSelecting: _isMultiSelecting,
+			isBlockInsertionPointVisible,
+			getBlockInsertionPoint,
+			getBlockOrder,
+		} = select( 'core/block-editor' );
+
+		const insertionPoint = getBlockInsertionPoint();
+		const order = getBlockOrder( insertionPoint.rootClientId );
+
+		return {
+			isMultiSelecting: _isMultiSelecting(),
+			isInserterVisible: isBlockInsertionPointVisible(),
+			selectedClientId: order[ insertionPoint.index ],
+			isNestedBlock: Boolean( insertionPoint.rootClientId ),
+		};
+	}, [] );
+
+	const isVisible =
+		Boolean( selectedClientId ) || isInserterForced || isInserterVisible;
+
+	const isInserterShown = Boolean( selectedClientId ) && ! isNestedBlock;
```

Comment:
Why is `! isNestedBlock` here? Why wouldn't we want to show the inserter in nested contexts?

Reply:
I removed it because I was testing with button group and figured there is an inserter at the end so why have another one.
![image](https://user-images.githubusercontent.com/17088041/93764685-36fc2700-fc1c-11ea-9be3-b8366c97f53b.png)
That wasn't the right approach. I've removed this logic.